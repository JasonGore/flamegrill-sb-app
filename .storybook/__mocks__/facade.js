console.log('__mocks__/facade');

// TODO: Don't think this approach is working with "new" storybook stories
export const storiesOf = function storiesOf() {
  var api = {};
  var story;
  console.log('storiesOf');
  api.add = (name, func)=> {
    console.log('add');
    story = func();
    snapshot(name, story);
    return api;
  };
  api.addWithInfo = (name, func)=> {
    console.log('addWithInfo');
    story = func();
    snapshot(name, story);
    return api;
  };
  return api;
};

export const action = () => {};

export const linkTo = () => {};

export const specs = (spec) => {
  console.log('spec');
  spec()
};

export const snapshot = (name, story) => {
  console.log('snapshot');
  it(name, function () {
    let renderer = require("react-test-renderer");
    const tree = renderer.create(story).toJSON();
    expect(tree).toMatchSnapshot();
  });
};

export const describe = jasmine.currentEnv_.describe;
export const it = jasmine.currentEnv_.it;
