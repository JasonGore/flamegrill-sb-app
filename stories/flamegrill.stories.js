import React from 'react';

import { mount } from 'enzyme';
import expect from 'expect';
import { action } from '@storybook/addon-actions';

// import { storiesOf } from '@storybook/react';
// import { specs, describe, it } from 'storybook-addon-specifications';

import { storiesOf, specs, describe, it } from '../.storybook/facade';

export default {
  title: 'Flamegrill',
};

const stories = storiesOf('Button', module);

stories.add('Hello World', function () {
  const story =
    <button onClick={action('Hello World')}>
      Hello World
    </button>;

  specs(() => describe('Hello World', function () {
    it('Should have the Hello World label', function () {
      let output = mount(story);
      expect(output.text()).toContain('Hello World');
    });
  }));

  return story;
});

export const SampleSpec = () => {
  const story =
    <button onClick={action('Hello World')}>
      Hello World
    </button>;

  specs(() => describe('Hello World', function () {
    it('Should have the Hello World label', function () {
      let output = mount(story);
      expect(output.text()).toContain('Hello World');
    });
  }));

  return story;
};
