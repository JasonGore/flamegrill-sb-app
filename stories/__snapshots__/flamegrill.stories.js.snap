// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello World 1`] = `
<button
  onClick={[Function]}
>
  Hello World
</button>
`;
